core/drivers/firewall/stm32_risaf.c

```diff
@@ -555,10 +557,136 @@ stm32_risaf_pm(enum pm_op op, unsigned int pm_hint,
 	return res;
 }
 
+static TEE_Result stm32_risaf_acquire_access(struct firewall_query *fw,
+					     paddr_t paddr, size_t size,
+					     bool read, bool write)
+{
+	struct stm32_risaf_instance *risaf = NULL;
+	struct stm32_risaf_region *region = NULL;
+	uint32_t cidcfgr = 0;
+	unsigned int i = 0;
+	uint32_t cfgr = 0;
+	vaddr_t base = 0;
+	uint32_t id = 0;
+
+	assert(fw->ctrl->priv && (read || write));
+
+	if (paddr == TZDRAM_BASE && size == TZDRAM_SIZE)
+		return TEE_SUCCESS;
+
+	risaf = fw->ctrl->priv;
+	base = risaf_base(risaf);
+
+	if (fw->arg_count != 1)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	/*
+	 * RISAF region configuration, we assume the query is as
+	 * follows:
+	 * firewall->args[0]: Region configuration
+	 */
+	id = _RISAF_GET_REGION_ID(fw->args[0]);
+
+	if (!id || id >= risaf->pdata.nregions)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	for (i = 0; i < risaf->pdata.nregions; i++) {
+		if (id == _RISAF_GET_REGION_ID(risaf->pdata.regions[i].cfg)) {
+			region = &risaf->pdata.regions[i];
+			if (region->addr != paddr || region->len != size)
+				return TEE_ERROR_GENERIC;
+			break;
+		}
+	}
+	if (!region)
+		return TEE_ERROR_ITEM_NOT_FOUND;
```

Comment:
Suggestion: could factorize get the `region` from the config `id`. (lines 588..602 and 651..663)

Reply:
I'll keep as-is and reconsider it if further APIs are necessary